File: src/main/java/top/wcpe/wcbot/entity/ServerData.java
package top.wcpe.wcbot.entity;

import cn.nukkit.utils.Config;
import lombok.Data;

import java.util.List;

/**
 * 服务器数据存储
 *
 * @author: WCPE 1837019522@qq.com
 * @create: 2021-06-22 21:43
 */
@Data
public class ServerData {
    private List<Long> enableQQGroup;

    public ServerData(Config cfg) {
        this.enableQQGroup = cfg.getLongList("enableQQGroup");
    }

    public Config saveToConfig() {
        Config cfg = new Config();
        cfg.set("enableQQGroup", this.enableQQGroup);
        return cfg;
    }
}

File: src/main/kotlin/top/wcpe/wcbot/entity/ServerData.kt
package top.wcpe.wcbot.entity

import cn.nukkit.utils.Config
import top.wcpe.wcbot.WCBot
import java.io.File

/**
 * 由 WCPE 在 2021/12/2 19:40 创建
 *
 * Created by WCPE on 2021/12/2 19:40
 *
 * Github: https://github.com/wcpe
 *
 * QQ: 1837019522
 *
 * @author WCPE
 */
data class ServerData(
    var ownerQQ: Long? = null,
    var enableQQGroup: MutableList<Long> = mutableListOf()
) {
    constructor() : this(null) {
        load()
    }

    fun load() {
        val cfg = Config(file)
        this.ownerQQ = cfg.getLong("ownerQQ")
        this.enableQQGroup = cfg.getLongList("enableQQGroup")
    }

    private val file = File(WCBot.instance.dataFolder, "serverData.yml")

    fun save(): Boolean {
        val cfg = Config(file)
        cfg["ownerQQ"] = ownerQQ
        cfg["enableQQGroup"] = enableQQGroup
        return cfg.save()
    }
}